File: Sorting/python/Insertion_sort.py
# Program to Implement Insertion Sort

# Input : 3 1 2 5 4
# Output :  1, 2, 3, 4, 5

import time

###---------------  Function to implement Insertion sort ----------------###
def insertion_sort(array):
    for i in range(1, len(array)):
        for j in range(i-1, -1, -1):
            if array[j] > array[j+1]:
                array[j], array[j+1] = array[j+1], array[j]
            
            else:
                break
    return array

###------------------------ driver function ----------------------###
def main():
    print()

    array = [ int(i) for i in input("Enter the elements in array : ").strip().split(" ")]
    
    print("---- Using Insertion Sort -----")
    start = time.time()
    array_ins = insertion_sort(array)
    end = time.time()
    total_time_insertion = end - start
    print("Sorted Array :", array_ins)
    print(f"Total time taken by program is {total_time_insertion}")
    
###-------------- Calling driver function ------------------###
main()

File: Sorting/InsertionSort.c
#include<stdio.h>

int main() {
   int i, j, n, temp, a[20];

   printf("enter the no of elements you want to sort ");
   scanf("%d", &n);

   printf("enter the elements: ");
   for (i = 0; i < n; i++) {
      scanf("%d", &a[i]);
   }

   for (i = 1; i < n; i++) {
      temp = a[i];
      j = i - 1;
      while ((temp < a[j]) && (j >= 0)) {
         a[j + 1] = a[j];
         j = j - 1;
      }
      a[j + 1] = temp;
   }

   printf("Sorted array is ");
   for (i = 0; i < n; i++) {
      printf(" %d ", a[i]);
   }
return 0;
}
